src/freenet/pluginmanager/PluginManager.java

```diff
@@ -1706,39 +1559,35 @@ public static void setLanguage(LANGUAGE lang) {
 	}
 
 	private void setPluginLanguage(final LANGUAGE lang) {
-		synchronized(pluginWrappers) {
-			for(PluginInfoWrapper pi: pluginWrappers) {
-				if(pi.isL10nPlugin()) {
-					final FredPluginL10n plug = (FredPluginL10n)(pi.plug);
-					executor.execute(new Runnable() {
-						@Override
-						public void run() {
-							try {
-								plug.setLanguage(lang);
-							} catch (Throwable t) {
-								Logger.error(this, "Cought Trowable in Callback", t);
-							}
-						}}, "Callback");
-				} else if(pi.isBaseL10nPlugin()) {
-					final FredPluginBaseL10n plug = (FredPluginBaseL10n)(pi.plug);
-					executor.execute(new Runnable() {
-						@Override
-						public void run() {
-							try {
-								plug.setLanguage(lang);
-							} catch (Throwable t) {
-								Logger.error(this, "Cought Trowable in Callback", t);
-							}
-						}}, "Callback");
-				}
+		for (PluginInfoWrapper pluginInfoWrapper : pluginWrappers) {
+			if (pluginInfoWrapper.isL10nPlugin()) {
+				final FredPluginL10n plug = (FredPluginL10n) (pluginInfoWrapper.plug);
+				executor.execute(new Runnable() {
+					@Override
+					public void run() {
+						try {
+							plug.setLanguage(lang);
+						} catch (Throwable t) {
+							Logger.error(this, "Cought Trowable in Callback", t);
+						}
+					}
+				}, "Callback");
+			} else if (pluginInfoWrapper.isBaseL10nPlugin()) {
+				final FredPluginBaseL10n plug = (FredPluginBaseL10n) (pluginInfoWrapper.plug);
+				executor.execute(new Runnable() {
+					@Override
+					public void run() {
+						try {
+							plug.setLanguage(lang);
+						} catch (Throwable t) {
+							Logger.error(this, "Cought Trowable in Callback", t);
+						}
+					}
+				}, "Callback");
 			}
 		}
 	}
 
-	public THEME getFProxyTheme() {
```

Comment:
Okay, though I think that plugins that ask the plugin manager for the current fproxy theme should be taken out back and shot. Should we schedule a date for removal? In six months, maybe? A year?